test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,314 @@
+
+	createFqdnPolicyInNamespace(t, data, testFQDN, "test-anp-fqdn", "custom-dns", "fqdn-cache-test")
+	createToolboxPod(t, data, dnsServiceIP)
+	curlTarget := testFQDN
+	curlFQDN := func() (string, error) {
+		cmd := []string{"curl", curlTarget}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, toolboxPodName, toolboxContainerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), toolboxPodName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlFQDN()
+		assert.NoError(t, err)
+		t.Logf("Response of curl to FQDN - %s", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the FQDN : ", testFQDN)
+
+	// confirm that the FQDN resolves to the expected IP address and store it to simulate caching of this IP associated with the FQDN.
+	t.Logf("Resolving FQDN to simulate caching the current IP inside Toolbox pod...")
+	resolvedIP, err := data.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFQDN, false, dnsServiceIP)
+	fqdnIp := resolvedIP.String()
+	require.NoError(t, err, "failed to resolve FQDN to an IP from toolbox pod")
+	require.Equalf(t, agnHostPodOneIP, fqdnIp, "The IP set against the FQDN in the DNS server should be the same, but got %s instead of %s", fqdnIp, agnHostPodOneIP)
+	t.Logf("Successfully received the expected IP %s using the dig command against the FQDN", fqdnIp)
+
+	// update the IP address mapped to the FQDN in the custom DNS ConfigMap.
+	t.Logf("Updating host mapping in DNS server config to use new IP : %s", agnHostPodTwoIP)
+	customDnsConfigMap.Data = createDnsConfig(t, map[string]string{agnHostPodTwoIP: testFQDN}, dnsTTL)
+	require.NoError(t, data.UpdateConfigMap(customDnsConfigMap), "failed to update configmap with new IP")
+	t.Logf("Successfully updated dns configMap with new IP : %s", agnHostPodTwoIP)
+
+	// try to trigger an immediate refresh of the configmap by setting annotations in custom DNS server pod, this way
+	// we try to bypass the kubelet sync period which may be as long as (1 minute by default) + TTL of ConfigMaps.
+	// Ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
+	require.NoError(t, data.setPodAnnotation(data.testNamespace, "custom-dns-server", "test.antrea.io/random-value",
+		randSeq(8)), "failed to update custom dns pod annotation.")
+
+	curlTarget = fqdnIp
+	// finally verify that Curling the previously cached IP fails after DNS update.
+	assert.EventuallyWithT(t, func(collectT *assert.CollectT) {
+		t.Logf("Trying to curl the existing cached IP of the domain - %s", fqdnIp)
+		stdout, err := curlFQDN()
+		if err != nil {
+			t.Logf("The test failed because of error :  %+v", err)
+		} else {
+			t.Logf("Response of curl to cached IP - %+v", stdout)
```

Comment:
```suggestion
			t.Logf("Response of curl to cached IP: %s", stdout)
```

`%+v` is used to print structs